include/swift/SIL/SILValue.h

```diff
@@ -326,15 +326,14 @@ class Operand {
   /// FIXME: this could be space-compressed.
   SILInstruction *Owner;
 
+#warning "ask john about making these constructors private"
+public:
   Operand(SILInstruction *owner) : Owner(owner) {}
   Operand(SILInstruction *owner, SILValue theValue)
       : TheValue(theValue), Owner(owner) {
     insertIntoCurrent();
   }
```

Comment:
@rjmccall Do you have any qualms about making these `Operand` constructors public? It felt easier for me to make `KeyPathInst` keep its own `TrailingObjects`-based list of operands than to try to use `TailAllocatedOperandList`, since the instruction already had trailing `Substitution`s.